src/Okta.Sdk.IntegrationTests/GroupScenarios.cs

```diff
@@ -507,5 +509,650 @@ public async Task RemoveGroupTargetFromGroupAdministratorRoleGivenToGroup()
+                {
+                    People = new GroupRulePeopleCondition
+                    {
+                        Users = new GroupRuleUserCondition
+                        {
+                            Exclude = new List<string> {createdUser.Id},
+                        },
+                        Groups = new GroupRuleGroupCondition
+                        {
+                            Exclude = new List<string>(),
+                        },
+                    },
+                    Expression = new GroupRuleExpression
+                    {
+                        Value = "user.countryCode==\"US\"",
+                        Type = "urn:okta:expression:1.0",
+                    },
+                },
+                Actions = new GroupRuleAction
+                {
+                    AssignUserToGroups = new GroupRuleGroupAssignment
+                    {
+                        GroupIds = new List<string>{createdGroup.Id},
+                    },
+                }, 
+            };
+
+            var createdGroupRule = await testClient.Groups.CreateGroupRuleAsync(groupRule);
+            try
+            {
+                createdGroupRule.Should().NotBeNull();
+                createdGroupRule.Id.Should().NotBeNullOrEmpty();
+                createdGroupRule.Type.Should().Be(groupRule.Type);
+                createdGroupRule.Name.Should().Be(groupRule.Name);
+            }
+            finally
+            {
+                await testClient.Groups.DeactivateGroupRuleAsync(createdGroupRule.Id);
+                await testClient.Groups.DeleteGroupRuleAsync(createdGroupRule.Id);
+                await createdUser.DeactivateAsync();
+                await createdUser.DeactivateOrDeleteAsync();
+                await createdGroup.DeleteAsync();
+            }
+        }
+
+        [Fact]
+        public async Task UpdateGroupRule()
+        {
+            var testClient = TestClient.Create();
+            var guid = Guid.NewGuid();
+            var testNickName = $"dotnet-sdk-{nameof(UpdateGroupRule)}";
+            var testUserName = $"{testNickName}-{guid}@example.com";
+
+            var profile = new UserProfile
+            {
+                FirstName = "John",
+                LastName = $"{nameof(UpdateGroupRule)}",
+                Email = testUserName,
+                Login = testUserName,
+                ["nickName"] = testNickName,
+            };
+
+            var createdUser = await testClient.Users.CreateUserAsync(new CreateUserWithPasswordOptions
+            {
+                Profile = profile,
+                Password = "Abcd1234",
+            });
+
+            var createdGroup = await testClient.Groups.CreateGroupAsync(new CreateGroupOptions
+            {
+                Name = $"dotnet-sdk:{nameof(UpdateGroupRule)}-{guid}",
+                Description = $"dotnet-sdk:{nameof(UpdateGroupRule)}-{guid}",
+            });
+
+            Thread.Sleep(3000);
```

Comment:
Same comment here. Do we need this?

Reply:
Experienced intermittent errors due to attempting to read data that wasn't fully committed.  This helped to alleviate the issues I was seeing.